app/Services/Pocket.php

```diff
@@ -0,0 +1,221 @@
+<?php namespace App\Services;
+
+use \App\Items as ItemModel;
+use \App\UserItems as UserItemsModel;
+use GuzzleHttp\Exception\ClientException;
+use Goose\Client as GooseClient;
+use Readability;
+use \Log;
+
+class Pocket {
+
+    const POCKET_API_URL = "https://getpocket.com/v3/";
+
+    /**
+     * User to make pocket requests on behalf of.
+     * @var
+     */
+    protected $user;
+
+    /**
+     * @var \GuzzleHttp\Client $httpClient
+     */
+    protected $httpClient;
+
+    public function __construct($user)
+    {
+        $this->user = $user;
+    }
+
+    /**
+     * Add latest pock
+     *
+     * @throws \Exception
+     */
+    public function synchronisePocketItems()
+    {
+
+        $feed = $this->getPocketFeed();
+
+        $urls = array();
+        foreach ($feed as $feedItem) {
+            $item = $this->addItem($feedItem['resolved_url'], $feedItem['resolved_title'], $feedItem['excerpt']);
+            if ($item) {
+                $this->addUserItem($item);
+                $urls[] = $feedItem['resolved_url'];
+            }
+        }
+
+        // Remove previous items which are no longer in the feed
+        $affectedRows = UserItemsModel::whereNotIn('item_id', array_map("md5", $urls))->delete();
+    }
+
+
+
+    /**
+     * @return array|mixed
+     */
+    protected function getPocketFeed()
+    {
+
+        try {
+            $response = $this->getHttpClient()->post("get", array(
+                'headers' => [
+                    'Content-Type' => 'application/json',
+                    'X-Accept' => 'application/json'
+                ],
+                "body" => json_encode(
+                    array(
+                        "consumer_key" => env('POCKET_API'),
+                        "access_token" => $this->user->access_token,
+                        "contentType"  => "article",
+                        "status"       => "unread",
+                        "sort"         => "newest",
+                        "detailType"   => "simple"
+                    )
+                )
+            ));
+
+        } catch (\Exception $e) {
+            Log::error($e->getMessage());
+            return array();
+        }
+
+        $statusCode = $response->getStatusCode();
+        if ($statusCode != 200) {
+            if ($statusCode == 400) {
+                Log::error('Invalid pocket request');
+            } elseif($statusCode == 401) {
+                Log::error('Problem authenticating user');
+            } elseif($statusCode == 403) {
+                Log::error('Authenticated but lack of permission to access this resource');
+            } elseif($statusCode == 500) {
+                Log::error('Pocket server issue');
+            }
```

Comment:
You log an error on an unsuccessful response, but still proceed to read json afterwards. Should probably return here?